src/maxKey.js

```diff
@@ -8,13 +8,33 @@
  * @param {object} array - A non-empty array of numbers
  * @returns {number} The key with the largest value
  */
-function maxKey(obj) {
-  // This is your job. :)
+
+ let a = 22
+let b = 44
+let c = 13
+let d = 3
+
+function maxKey(array) {
+
+  //   let keyValues = {
+// "a": "22",
+// "b": "44",
+// "c": "13",
+// "d": "3"
+// }
+ let maxKeySoFar = array[0];
+ for (let element of array) {
+   if (element > maxKeySoFar) {
+   maxKeySoFar = element;
+ }
+}
+return maxKeySoFar;
 }
 
 if (require.main === module) {
   console.log('Running sanity checks for maxKey:');
-
+console.log('-------');
+console.log(maxKey([a,b,c,d]));
```

Comment:
If that's your intention then make it clear to me by, say, leaving a comment in the code or in the PR telling me that's what you're doing. If that's not your intention and you're misunderstanding what the exercise is asking, make sure to read the comment and the provided example input/output.

You always have to be clear on at least two things when writing a function: what type of data does the function expect as input and what type of data is it returning?

In this case, the function is taking an object as input and returning a string. You've rewritten it to take an array as input and return a number.

Although re-reading the comment, I see I said it returns a `{number}` not `{string}`. That's my mistake, but the rest of the comment describes clearly what the function is meant to do in both English and an example in code.

In fact, the example(s) from the comment translate directly into sanity checks.

From the comment:

```javascript
maxKey({a: 10, b: -4, c: 20}); // => 'c'
```

As a sanity check:

```javascript
let actualValue = maxKey({a: 10, b: -4, c: 20});
let expectedValue = 'c';
console.log(actualValue === expectedValue, 'Expecting', expectedValue, 'got', actualValue);
```